src/frequenz/client/dispatch/test/_service.py

```diff
@@ -50,12 +51,34 @@
 class FakeService:
     """Dispatch mock service for testing."""
 
+    @dataclass(frozen=True)
+    class StreamEvent:
+        """Event for the stream."""
+
+        microgrid_id: int
+        """The microgrid id."""
+
+        event: DispatchEvent
+        """The event."""
+
+    _stream_sender: Sender[StreamEvent] = dataclasses.field(init=False)
+    """Sender for dispatch events."""
+
+    _stream_channel: Broadcast[StreamEvent] = dataclasses.field(
+        default_factory=lambda: Broadcast(name="fakeservice-dispatch-stream")
+    )
+    """Channel for dispatch events."""
+
     dispatches: dict[int, list[Dispatch]] = dataclasses.field(default_factory=dict)
     """List of dispatches per microgrid."""
 
     _last_id: int = 0
     """Last used dispatch id."""
 
+    def __post_init__(self) -> None:
+        """Initialize the stream sender."""
+        self._stream_sender = self._stream_channel.new_sender()
```

Comment:
To be honest I don't see a point on having `FakeService` be a `dataclass` at this point, it is causing more harm than good. You have one public property and 3 property hacks. Why not making it a regular class with a regular `__init__()`?